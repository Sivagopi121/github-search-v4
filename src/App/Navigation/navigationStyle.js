import styled from '@emotion/styled';

export const NavButtonsContainer = styled.div`
    overflow: hidden;
    position: fixed;
    top: 0;
    width: 100%;
    z-index: 1;
    background-color: #1a62ab;
    display: flex;
    align-items: baseline;

    @media (max-width: 480px) {
        flex-direction: column;
        justify-content: center;
        align-items: center;
      }
`;

export const NavigationLink = styled.div`
    font-size: 12px;
    letter-spacing: 3.5px;
    font-weight: 500;
    text-transform: uppercase;
    padding: 20px;
    text-decoration: none;
    a {
        color: #ffffff;
    }
    @media (max-width: 480px) {
        padding: 10px;
    }
`;

export const NavigationSearch = styled.div`
padding: 0 10px;
@media (max-width: 480px) {
    padding: 10px 10px;
`;
