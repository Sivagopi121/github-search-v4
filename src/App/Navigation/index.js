import React from 'react';
import { Link, withRouter } from 'react-router-dom';

import * as routes from '../../constants/routes';
import Button from '../../Button';
import Input from '../../Input';

import { NavButtonsContainer, NavigationLink, NavigationSearch } from './navigationStyle';

const Navigation = ({
  location: { pathname },
  organizationName,
  onOrganizationSearch,
}) => (
  <NavButtonsContainer>
    <NavigationLink>
      <Link to={routes.PROFILE}>Profile</Link>
    </NavigationLink>
    <NavigationLink>
      <Link to={routes.ORGANIZATION}>Organization</Link>
    </NavigationLink>

    {pathname === routes.ORGANIZATION && (
      <OrganizationSearch
        organizationName={organizationName}
        onOrganizationSearch={onOrganizationSearch}
      />
    )}
  </NavButtonsContainer>
);

class OrganizationSearch extends React.Component {
  state = {
    value: this.props.organizationName,
  };

  onChange = (event) => {
    this.setState({ value: event.target.value });
  };

  onSubmit = (event) => {
    this.props.onOrganizationSearch(this.state.value);

    event.preventDefault();
  };

  render() {
    const { value } = this.state;

    return (
      <NavigationSearch>
        <form onSubmit={this.onSubmit}>
          <Input
            color="white"
            type="text"
            value={value}
            onChange={this.onChange}
          />
          {' '}
          <Button color="white" type="submit">
            Search
          </Button>
        </form>
      </NavigationSearch>
    );
  }
}

export default withRouter(Navigation);
