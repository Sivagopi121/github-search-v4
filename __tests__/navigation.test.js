import React from 'react';
import renderer from 'react-test-renderer';
import Navigation from '../src/App/Navigation/index'
import { MemoryRouter } from 'react-router-dom';


test('Error renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Navigation/>
      </MemoryRouter>
      )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
