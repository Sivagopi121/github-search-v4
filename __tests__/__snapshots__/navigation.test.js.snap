// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders correctly 1`] = `
<div
  className="css-19wvdzc-NavButtonsContainer es2a6wv0"
>
  <div
    className="css-1fjgb5z-NavigationLink es2a6wv1"
  >
    <a
      href="/profile"
      onClick={[Function]}
    >
      Profile
    </a>
  </div>
  <div
    className="css-1fjgb5z-NavigationLink es2a6wv1"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Organization
    </a>
  </div>
  <div
    className="css-1tgob8f-NavigationSearch es2a6wv2"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        className="Input Input_white"
        onChange={[Function]}
        type="text"
      />
       
      <button
        className="undefined Button Button_white"
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</div>
`;
